exporters/metric/dogstatsd/aggregation/distribution/distribution.go

```diff
@@ -0,0 +1,194 @@
+package distribution
+
+// lifted from import "go.opentelemetry.io/otel/sdk/metric/aggregator/array"
+
+import (
+	"context"
+	"math"
+	"sort"
+	"sync"
+	"unsafe"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/metric"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+type (
+	Aggregator struct {
+		// ckptSum needs to be aligned for 64-bit atomic operations.
+		ckptSum    core.Number
+		lock       sync.Mutex
+		current    points
+		checkpoint points
+	}
+
+	points []core.Number
+)
+
+var _ export.Aggregator = &Aggregator{}
+var _ aggregator.MinMaxSumCount = &Aggregator{}
+var _ aggregator.Distribution = &Aggregator{}
+var _ aggregator.Points = &Aggregator{}
+
+type Distribution interface {
+	AsDistribution() bool
```

Comment:
my go foo failed me on this one if you have a better approach?

Reply:
I don't see where this function is actually used anywhere. It seems in the exporter, you could just do your type assertions on `distribution.Aggregator` and it would still evaluate as you are intending it

Reply:
```if _, ok := agg.(distribution.Aggregator); ok { ... }``` 

causes:  impossible type assertion: distribution.Aggregator does not implement "go.opentelemetry.io/otel/sdk/export/metric".Aggregator (Checkpoint method has pointer receiver)

Reply:
ahh, `agg.(*distribution.Aggregator)` works

Reply:
Ah yeah. that one :D